### ISSUE TITLE
libavutil/common.h:179: error: ‘UINT64_C’

### ISSUE THREAD

**Issue by mmyself** [AutonomyLab/ardrone_autonomy#1]
i run in Ubuntu 10.04 64bit in this little issue by rosmake ardrone_autonomy

  In file included from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/FFMPEG/Includes/libavutil/avutil.h:125,
                   from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/VP_SDK/VP_Api/vp_api_picture.h:16,
                   from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/VLIB/video_controller.h:5,
                   from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/VLIB/video_codec.h:4,
                   from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/Soft/Common/config_keys.h:29,
                   from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/Soft/Common/ardrone_api.h:739,
                   from /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/Soft/Lib/ardrone_tool/ardrone_tool.h:4,
                   from /home/jan/ros/stacks/ardrone_autonomy/src/ardrone_sdk.h:25,
                   from /home/jan/ros/stacks/ardrone_autonomy/src/ardrone_driver.h:8,
                   from /home/jan/ros/stacks/ardrone_autonomy/src/ardrone_driver.cpp:1:
  /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/FFMPEG/Includes/libavutil/common.h: In function ‘int32_t av_clipl_int32_c(int64_t)’:
  /home/jan/ros/stacks/ardrone_autonomy/ARDroneLib/FFMPEG/Includes/libavutil/common.h:179: error: ‘UINT64_C’ was not declared in this scope

solved it with this small code, found by google
# ifndef INT64_C
# define INT64_C(c) (c ## LL)
# define UINT64_C(c) (c ## ULL)
# endif

wonder why. because the SDK2.0 compile in this system well, also the examples. 
maybe a make flag thing in ros???

**Comment by mani-monaj** [comment]
Hi,

Thank you for reporting this. Could you please tell me where did you put the above changes? 

> wonder why. because the SDK2.0 compile in this system well, also the examples.
> maybe a make flag thing in ros???

The original `Makefile` provided by the SDK does not build all the libraries in standalone mode. Therefore we wrote a basic Makfile which is wrapped by `build_sdk.sh`. I could not find all the flags needed because of the complex hierarchy of the SKD2.0 build system.

**Comment by mmyself** [comment]
Hi, i put in the line 40.
after the include and before the define section.

**Comment by mani-monaj** [comment]
> Hi, i put in the line 40.
> after the include and before the define section.

Can you please specify the filename?

**Comment by mmyself** [comment]
sorry for this. seems so clear for me ;)

libavutil/common.h

**Comment by mmyself** [comment]
back on the computer.
after updating to the new code i got the same issue, like before.

The better solution is this and it makes more sense.

#ifndef UINT64_C
#define UINT64_C(c) (c ## ULL)
#endif

**Comment by mani-monaj** [comment]
Could you please do this on a fresh code clone and check if it will fix the problem?

Add your proposed changes 

``` c
#ifndef UINT64_C
#define UINT64_C(c) (c ## ULL)
#endif
```

to file `ardrone_sdk.h`  before (or after) `#include <stdint.h>`

**Comment by mmyself** [comment]
Thats it, works well. 

```
diff --git a/src/ardrone_sdk.h b/src/ardrone_sdk.h
index 5519de4..221e497 100644
--- a/src/ardrone_sdk.h
+++ b/src/ardrone_sdk.h
@@ -17,6 +17,11 @@
 #endif
 #include <stdint.h>

+
+#ifndef UINT64_C
+#define UINT64_C(c) (c ## ULL)
+#endif
+
 extern "C" {

 #include <config.h>
```

**Comment by mani-monaj** [comment]
Thanks for testing that. I applied your changes to the code and will push that to github ASAP.

**Comment by mani-monaj** [comment]
Fixed in 337555f43b03aca8e19386c09795240c41e3632c

**Comment by hydraxman** [comment]
mark this
